### PR TITLE
Update PHP Parser

### PR DESCRIPTION
**Pull request by tcitworld**
Upgrade `nikic/php-parser` and `jeremeamia/SuperClosure`.

### CODE REVIEW

File: composer/autoload_files.php [hunk 1]
```diff
@@ -18,8 +18,8 @@
     '37a3dc5111fe8f707ab4c132ef1dbc62' => $vendorDir . '/guzzlehttp/guzzle/src/functions_include.php',
     'ebdb698ed4152ae445614b69b5e4bb6a' => $vendorDir . '/sabre/http/lib/functions.php',
     'b067bc7112e384b61c701452d53a14a8' => $vendorDir . '/mtdowling/jmespath.php/src/JmesPath.php',
-    'fe1bcd0336136e435eaf197895daf81a' => $vendorDir . '/nikic/php-parser/lib/bootstrap.php',
     'ad155f8f1cf0d418fe49e248db8c661b' => $vendorDir . '/react/promise/src/functions_include.php',
+    'bd9634f2d41831496de0d3dfe4c94881' => $vendorDir . '/symfony/polyfill-php56/bootstrap.php',
```
Comment (MorrisJobke) [inline composer/autoload_files.php]: Isn't this removed?

File: composer/installed.json [hunk 2]
```diff
@@ -3554,6 +3560,118 @@
             "shim"
         ]
     },
+    {
+        "name": "symfony/polyfill-php56",
```
Comment (MorrisJobke) [inline composer/installed.json]: This should not be in here. We are PHP 7.0+ and master is even 7.1+

Comment (tcitworld) [reply composer/installed.json]: Yup, composer pulled other changes for some reason. I don't get why https://github.com/nextcloud/3rdparty/blob/master/composer.json#L10 was ignored.

**Comment by danielkesselberg** [comment]
Superseded by https://github.com/nextcloud/3rdparty/pull/223?

**Comment by nickvergessen** [comment]
Yeah, sorry didnt see this one

**Comment by nickvergessen** [comment]
However I removed SuperClosure, because I didn't find it being used anywhere anymore.

**Pull request closed without merging**